tests/jerry/es2015/typedarray-prototype-indexof.js

```diff
@@ -0,0 +1,64 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var array =  new Uint8Array([0, 1, 2, 3, 0, 5, 6]);
+
+var index = array.indexOf(0);
+assert(index === 0);
+assert(array[index] === 0);
+
+assert(array.indexOf(0, 1) === 4);
+assert(array.indexOf(0, 5) === -1);
+
+var index = array.indexOf(6);
+assert(index === 6);
+assert(array[index] === 6);
+
+assert(array.indexOf(6, 7) === -1);
+
+var index = array.indexOf(3);
+assert(index === 3);
+assert(array[index] === 3);
+
+assert(array.indexOf(0, NaN) === 0);
+assert(array.indexOf(0, Infinity) === 0);
+assert(array.indexOf(3, -Infinity) === 3);
+
+assert([true].indexOf(true, -0) === 0);
```

Comment:
AFAIK, this is not typedarray indexOf call. The `[true]` here is a simple array.